src/status_im/mailserver/core.cljs

```diff
@@ -276,9 +276,28 @@
     (log/debug "Adjusting mailserver request" "from:" from "adjusted-from:" adjusted-from)
     adjusted-from))
 
-(fx/defn handle-request-success [{:keys [db]} {:keys [request-id]}]
+(fx/defn handle-request-success [{{:keys [chats] :as db} :db}
+                                 {:keys [request-id topics]}]
   (when (:mailserver/current-request db)
-    {:db (assoc-in db [:mailserver/current-request :request-id] request-id)}))
+    (let [fresh-public-chats    (into {}
```

Comment:
there's a bit of duplication, you could refactor in a fn such as `chats->fresh-chats chats` or at least factor out the transducer `(comp (f....))`

Reply:
got it. will do.

Reply:
done